### DISCUSSION TITLE
Python API - copy dataset

### DISCUSSION THREAD

**Discussion by smidm** [openvinotoolkit/datumaro#1190] [Q&A] [answered]
How to copy or clone a dataset? I would like to use `Dataset.filter`, which is done in-place and continue working with the original dataset.

**Comment by vinnamkim** [comment, marked as answer]
Hi @smidm, You can use `deepcopy` such as

```python
import datumaro as dm
from copy import deepcopy

dataset = dm.Dataset.import_from(...)
cloned = deepcopy(dataset)
```

**Reply by smidm** [reply]
This is how I did it. I'm bit uncertain of the outcome when using `copy` module on complex objects. In many cases it fails when the object has db connections, file handles, ... I'd propose to add `copy()` or `clone()` method to the `Dataset` object to make it easier for users.

**Reply by vinnamkim** [reply]
Thanks, we will work on your suggestion.

**Comment by lvzhengcq** [comment]
If you want to filter a dataset using the Dataset.filter method in Python, and you also want to keep the original dataset intact, you need to create a copy or clone of the dataset before applying the filter. There are a few ways to achieve this, depending on the type of dataset you are working with.
Let's assume you are using a library like pandas for handling datasets:
import pandas as pd

# Assuming 'original_dataset' is your original dataset
original_dataset = pd.read_csv('your_dataset.csv')

# Create a copy of the dataset
filtered_dataset = original_dataset.copy()

# Apply the filter to the copied dataset
filtered_dataset = filtered_dataset.filter(your_condition)

# Continue working with both the original and filtered datasets

Replace 'your_dataset.csv' with the actual path to your dataset file, and your_condition with the filtering condition you want to apply.
If you are working with a different type of dataset or library, the concept is similar: create a copy of the dataset before applying any in-place operations. Different libraries might have different methods for copying datasets, so be sure to check the documentation of the specific library you are using.
Remember, creating a copy ensures that changes made to one dataset do not affect the other, allowing you to work with both the original and filtered datasets independently.

**Reply by smidm** [reply]
this is chatgpt generated content without context of `Dataset`

**Reply by vinnamkim** [reply]
Sorry but I think this answer can confuse our community users, so I hide it.
